sjsonnet/src/sjsonnet/Std.scala

```diff
@@ -491,29 +529,22 @@ private object Get extends Val.Builtin("get", Array("o", "f", "default", "inc_hi
 
   private object Split extends Val.Builtin2("split", "str", "c") {
     def evalRhs(_str: Val, _c: Val, ev: EvalScope, pos: Position): Val = {
-      val str = _str.asString
-      val cStr = _c.asString
-      if(cStr.length != 1) Error.fail("std.split second parameter should have length 1, got "+cStr.length)
-      val c = cStr.charAt(0)
-      val b = new mutable.ArrayBuilder.ofRef[Lazy]
-      var i = 0
-      var start = 0
-      while(i < str.length) {
-        if(str.charAt(i) == c) {
-          val finalStr = Val.Str(pos, str.substring(start, i))
-          b.+=(finalStr)
-          start = i+1
-        }
-        i += 1
-      }
-      b.+=(Val.Str(pos, str.substring(start, math.min(i, str.length))))
-      new Val.Arr(pos, b.result())
+      new Val.Arr(pos, _str.asString.split(java.util.regex.Pattern.quote(_c.asString), -1).map(s => Val.Str(pos, s)))
```

Comment:
This implementation will be slower than the original because it will need to internally allocate a fresh `Pattern` invocation on every iteration, whereas the old version avoided that.

I was actually considering trying to refactor `SplitLimit` to work closer to `Split`.

Alternatively, we might consider using specialization to cache the Pattern instances: I have a some work-in-progress draft code to do this for a couple of other built-ins.